### ISSUE TITLE
Break a block using player's held tool

### ISSUE THREAD

**Issue by SkymeRR** [SkriptLang/Skript#2397]
I think this would be a really good implementation for skript, specially for code optimization. I had to make a 300 line code to obtain an accurate result according to minecraft's mechanics
![image](https://user-images.githubusercontent.com/53543037/63726299-cafe9200-c854-11e9-97c3-fc386ee2a602.png)

Example: 
This code loops all blocks in radius 2 around player, and destroys them (break loop-block) but then I had to make the fortune and silk touch enchantment mechanics on skript for every single ore (using a lot of checks and "chance of X%")
![image](https://b.catgirlsare.sexy/yOU7.png)

With this implementation I could do it in 6 lines
![image](https://user-images.githubusercontent.com/53543037/63726220-88d55080-c854-11e9-8b15-8d36b5d63a3a.png)

**Comment by SkymeRR** [comment]
Alright, thank you for the feedback and for correcting me

**Comment by Blueyescat** [comment]
This is easily possible using NMS and not possible with Spigot like Shane said. It doesn't make any sense to list every conditions, nor Skript's job.

skript-mirror code (use 2.0.0-snapshot: https://circleci.com/gh/btk5h/skript-mirror/3#artifacts/containers/0)
yes it supports everything (?) because it literally forces the player to break the block, great!
```vb
effect:
	patterns:
		make %player% break %blocks% [(1¦without [break] sound[s])]
		force %player% to break %blocks% [(1¦without [break] sound[s])]
	trigger:
		loop exprs-2:
			if parse mark is not 1:
				# BlockSoundGroup requires Paper 1.14
				# Sound categories and advanced play sound effect requires Skript 2.4x
				play sound "%loop-value.getSoundGroup().getBreakSound()%" in blocks category with volume 0.3 at loop-value
			set {_interactManager} to expr-1.getHandle().playerInteractManager
			{_interactManager}.breakBlock(loop-value.getPosition())
```

**Comment by SkymeRR** [comment]
@Blueyescat Thanks for the response, that would be great if I understood mirror xD

**Comment by Blueyescat** [comment]
You don't need to understand because i gave the full code, you will put it in a script then you can use `make player break loop-block`

Note: don't use "target block" for everything please, event-block -- clicked-block

**Comment by SkymeRR** [comment]
@Blueyescat yeah I just figured it out, thank you so much. Also, I would like to learn skript mirror, where can I do it

**Comment by Blueyescat** [comment]
https://skript-mirror.gitbook.io/docs/v/2.x

**Comment by SkymeRR** [comment]
By the way that code infinitely repeats itself (even though I use stop after `make player break loop-block`)

Code: `effect:
	patterns:
		make %player% break %blocks% [(1¦without [break] sound[s])]
		force %player% to break %blocks% [(1¦without [break] sound[s])]
	trigger:
		loop exprs-2:
			if parse mark is not 1:
				# BlockSoundGroup requires Paper 1.14
				# Sound categories and advanced play sound effect requires Skript 2.4x
				play sound "%loop-value.getSoundGroup().getBreakSound()%" in blocks category with volume 0.3 at loop-value
			set {_interactManager} to expr-1.getHandle().playerInteractManager
			{_interactManager}.breakBlock(loop-value.getPosition())

on break:
	if player's tool's lore contains "Explosive":
		loop all blocks in radius 2 around event-block:
			create safe explosion of force 1 at event-block
			force player to break loop-block
			stop
`

**Comment by Blueyescat** [comment]
Well that is correct, need to some checks. Would be good with metadatas. Please come to [Skript Hub Discord](https://skripthub.net/discord) for further discussion.

**Comment by Blueyescat** [comment]
So this should be closed

**Comment by ShaneBeee** [comment]
Just an update on this, Im not sure when, but apparently Spigot fixed this.
I was just fiddling around on a test server and noticed it works.

ex: Looking at a diamond ore block, holding a diamond pickaxe I ran:
`!break target block using player's tool`
it dropped 1 diamond

then I gave myself a diamond pick of fortune 100, and did the same command, this time it dropped 80+ diamonds.

Im looking thru Spigot commits trying to figure out when it was fixed, I cant find it, but it seems to be fixed (in 1.14.4 obviously)
